src/components/Square.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react'
+
+//component
+const Square = ({
+  id,
+  currentTurn,
+  value,
+  disabled,
+  winningSquares,
+  handleClick,
+}: SquareProps) => {
+  const [stateValue, setValue] = useState<string>('')
+  const [isWinningSquare, setIsWinningSquare] = useState<boolean>(false)
+
+  /**
+   * function for checking if 'this' square is part of the set of
+   * winning squares, sets the state to true if it is
+   */
+  const checkIsWinningSquare = () => {
+    winningSquares.forEach((winningSquareId) => {
+      if (id === winningSquareId) setIsWinningSquare(true)
+    })
+
+    if (winningSquares.length < 3) {
+      setIsWinningSquare(false)
+    }
+  }
+
+  useEffect(() => {
+    setValue(value)
+    checkIsWinningSquare()
+  })
```

Comment:
You should have a depencies array on this useEffect or it'll be called every re-render. You only want to check the win condition when the game state changes yeah?

```suggestion
  }, [value])
```